Update: Add a Kdtree collection type, collect from Kdtree at a box

902D9 opened the pull request:
The circle may cause the points at the boundary to not be collected, so a method to collect the points inside the box is added.

![image](https://github.com/user-attachments/assets/33d3e089-ecc2-44a2-9cb2-1e6ac892aedc)
![image](https://github.com/user-attachments/assets/3e8ee157-c558-4a9c-83b0-fe114569c477)
![image](https://github.com/user-attachments/assets/df1943bc-ef6a-4b91-b5fe-518f3b7db09f)

![image](https://github.com/user-attachments/assets/cfdcaf2e-f75c-46c8-9ede-efbec5f24d94)

902D9 committed:
Update: Add a Kdtree collection type, collect from Kdtree at a box

nutti reviewed (commented):
Thank you @902D9 for the contribution!
This seems a great addition to this plugin.

"Box" collection method itself is good to me.
But the user interface needs to be improved.
I think using the `radius` parameter for the box method is not intuitive.
Instead, adding a new Blueprint function `Collect From Kdtree (Cube)` and changing the existing Blueprint function name to `Collect From Kdtree (Sphere)` would be better.
Also, `Collect From Kdtree (Cube)` should accept a edge length instead of radius.

902D9 committed:
Update: function improvement

902D9 commented:
Added a new function called `Collect From Kdtree (Box)` and changed the display name of the original function to `Collect From Kdtree (Sphere)`
![image](https://github.com/user-attachments/assets/acde74d6-2df8-4fa9-9a25-1407a5814c82)
![image](https://github.com/user-attachments/assets/4551cd00-7c1f-466b-aa8c-7ea6f9d8ce7a)

nutti commented on Kdtree/Source/Kdtree/Private/AsyncKdtreeBPLibrary.cpp:
```diff
@@ -155,3 +155,67 @@ void UAsyncKdtreeBPLibrary::CollectFromKdtreeAsync(const UObject* WorldContextOb
+class FCollectFromKdtreeBoxTask : public FNonAbandonableTask
+{
+public:
+	FCollectFromKdtreeBoxTask(const FCollectFromKdtreeTaskParams& InParams, const FBox& InBox) : Params(InParams), Box(InBox)
```
Please create `FCollectFromKdtreeBoxTaskParams` and includes a `Box` member variable instead of passing `FBox` as the `InBox` argument.

nutti commented on Kdtree/Source/Kdtree/Private/AsyncKdtreeBPLibrary.cpp:
```diff
@@ -155,3 +155,67 @@ void UAsyncKdtreeBPLibrary::CollectFromKdtreeAsync(const UObject* WorldContextOb
+
+	FORCEINLINE TStatId GetStatId() const
+	{
+		RETURN_QUICK_DECLARE_CYCLE_STAT(FCollectFromKdtreeTask, STATGROUP_ThreadPoolAsyncTasks);
```
```suggestion
		RETURN_QUICK_DECLARE_CYCLE_STAT(FCollectFromKdtreeBoxTask, STATGROUP_ThreadPoolAsyncTasks);
```

nutti commented on Kdtree/Source/Kdtree/Private/AsyncKdtreeBPLibrary.cpp:
```diff
@@ -155,3 +155,67 @@ void UAsyncKdtreeBPLibrary::CollectFromKdtreeAsync(const UObject* WorldContextOb
+	}
+};
+
+void UAsyncKdtreeBPLibrary::CollectFromKdtreeAsync_Box(const UObject* WorldContextObject, const FKdtree& Tree, const FBox Box,
```
```suggestion
void UAsyncKdtreeBPLibrary::CollectFromKdtreeAsyncBox(const UObject* WorldContextObject, const FKdtree& Tree, const FBox Box,
```

nutti commented on Kdtree/Source/Kdtree/Private/AsyncKdtreeBPLibrary.cpp:
```diff
@@ -155,3 +155,67 @@ void UAsyncKdtreeBPLibrary::CollectFromKdtreeAsync(const UObject* WorldContextOb
+	if (UWorld* World = GEngine->GetWorldFromContextObject(WorldContextObject, EGetWorldErrorMode::LogAndReturnNull))
+	{
+		FLatentActionManager& LatentManager = World->GetLatentActionManager();
+		if (LatentManager.FindExistingAction<FCollectFromKdtreeAction>(LatentInfo.CallbackTarget, LatentInfo.UUID) == nullptr)
```
```suggestion
		if (LatentManager.FindExistingAction<FCollectFromKdtreeBoxAction>(LatentInfo.CallbackTarget, LatentInfo.UUID) == nullptr)
```

nutti commented on Kdtree/Source/Kdtree/Private/KdtreeBPLibrary.cpp:
```diff
@@ -36,6 +36,16 @@ void UKdtreeBPLibrary::CollectFromKdtree(
 	}
 }
 
+void UKdtreeBPLibrary::CollectFromKdtree_Box(const FKdtree& Tree, const FBox Box, TArray<int>& Indices,
```
```suggestion
void UKdtreeBPLibrary::CollectFromKdtreeBox(const FKdtree& Tree, const FBox Box, TArray<int>& Indices,
```

nutti reviewed (changes requested):
@902D9 
Thank you for creating a separated function.
Please improve the code structure based on my suggestion.

902D9 committed:
Update: function improvement and bug fix

902D9 commented:
> Thank you for creating a separated function.
> Please improve the code structure based on my suggestion.

I have improved the code as you suggested.

nutti reviewed (changes requested):
@902D9 
Thank you. All code looks good to me.
Please delete `Kdtree/Config/DefaultKdtree.ini`. I think this file is unintentionally included.

902D9 commented:
> @902D9 Thank you. All code looks good to me. Please delete `Kdtree/Config/DefaultKdtree.ini`. I think this file is unintentionally included.

This file contains the redirection information for the above function renames. If the blueprint in the project uses the old named function, it can help it identify the renamed C++ function.
 If you don't want the plugin to include these redirect information, I can delete it.

nutti commented:
@902D9 

This PR is not released yet.
So, we can delete this file safely.

Commit:
Delete the redirection config file

902D9 commented:
Deleted

nutti commented:
@902D9 

Merged.
Thank you for your contribution!
